apps/total-typescript/src/pages/products/[slug].tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import {GetServerSideProps} from 'next'
+import type {CommerceProps} from '@skillrecordings/commerce-server/dist/@types'
+import {propsForCommerce} from '@skillrecordings/commerce-server'
+import {Element} from 'react-scroll'
+import Layout from 'components/app/layout'
+import {getToken} from 'next-auth/jwt'
+import Image from 'next/legacy/image'
+import {motion, useScroll, useTransform} from 'framer-motion'
+import {PricingTiers} from '../../path-to-purchase-react/product-tiers'
+import {getProductBySlug} from '../../path-to-purchase-react/products.server'
+
+const ProductPage: React.FC<React.PropsWithChildren<CommerceProps>> = ({
```

Comment:
this is "just the buy route" but uses a product slug to load something specific and should present like a page instead of a "buy" widget.

"buy" widget is part of it too, similar to a workshop that is for sale and also serves as a "hub" for navigating your purchased products